svc/cosmetics.mjs

```
@@ -1,75 +1,54 @@
-          item.item_id = Number(itemId);
-          const hero =
-            item.used_by_heroes &&
-            typeof item.used_by_heroes === 'object' &&
-            Object.keys(item.used_by_heroes)[0];
-          function insert(cb) {
-            // console.log(item);
-            return queries.upsert(
-              db,
-              'cosmetics',
-              item,
-              {
-                item_id: item.item_id,
-              },
-              cb
-            );
-          }
-          if (hero) {
-            item.used_by_heroes = hero;
-          }
-          // console.log(item);
-          if (!item.item_id) {
-            return cb();
-          }
-          if (item.image_inventory) {
-            const spl = item.image_inventory.split('/');
-            const iconname = spl[spl.length - 1];
-            return utility.getData(
-              {
-                url: utility.generateJob('api_item_icon', {
-                  iconname,
-                }).url,
-                noRetry: true,
-              },
-              (err, body) => {
-                if (err || !body || !body.result) {
-                  return cb();
-                }
-                item.image_path = body.result.path;
-                return insert(cb);
-              }
-            );
-          }
-          return insert(cb);
-        },
-        cb
-      );
+      if (!item.item_id) {
+        return;
+      }
+      if (item.image_inventory) {
+        const spl = item.image_inventory.split('/');
+        const iconname = spl[spl.length - 1];
+        if (icons[iconname]) {
+          item.image_path = icons[iconname];
+        }
+      }
+      await queries.upsertPromise(db, 'cosmetics', item, {
+        item_id: item.item_id,
+      });
     }
-  );
+
+    const funcs = Object.keys(itemData.items_game.items).map(
+      (i) => () => processItem(i)
+    );
+    await eachLimit(funcs, 10);
+  } catch (error) {
```

I think since we're at top level we can just not try catch here and let it crash. The process manager will restart it